### DISCUSSION TITLE
Currency.asset related question

### DISCUSSION THREAD

**Discussion by Ongjming** [SmartlyDressedGames/Unturned-3.x-Community#2845] [Q&A] [answered]
I have looked through (https://github.com/SmartlyDressedGames/U3-Docs/blob/master/Currency.md) and had also tried making custom NPC vendor to accept random vanilla supplies type item as currency.

Roughly the start of the .asset file looks like:
"Metadata"
{
	"GUID" "GUIDnumbers here"
	"Type" "SDG.Unturned.ItemCurrencyAsset, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null"
}
"Asset"
{
	"ValueFormat" "{0:N2} CurrencyName"
	"Entries"
	[
		{
			"Item"
			{
				"GUID" "ItemGUID number here"
			}
			"Value" "0.01"
		}
So on as it continue down various other values.

So here are the questions in an attempt to get to know this better:
Q1. There isn't anything I should change about the starting metadata right? Other than the unique GUID.
Q2. If I wanted the .asset to work, is there any specific place I should be placing that file in?
Q3. If the Items are all from different mods and I wanted this .asset to work as a mod itself together with these other mods (let's say in singleplayer, the .asset is within custom map with custom NPC vendor) what do I need to do? Where do I place the file in if there are file structure restrictions for the game to be able to read it?
EDIT: Q4. The name of this .asset file for currency doesn't really matter so long the vendor file reference this file's GUID right?

Thanks!

**Comment by iBowie** [comment, marked as answer]
1) Yes, only GUID should be affected.
2) From my experience, works from any place, as long as it's in Bundles folder
3) I think game doesn't check for cases when item is not present at the moment of loading, only in game, so anywhere should be fine.
4) It doesn't matter, but i personally would use unique file names to avoid even a small chance of name collision.

**Reply by SDGNelson** [reply]
Thanks iBowie, these answers are correct.

**Reply by Ongjming** [reply]
Thanks! Hmm just one more question that came up:
Q. If I want to set up decimal value currency, under the "value" "0.01" how many decimal places can I go up to?

**Reply by SDGNelson** [reply]
Value must be a positive integer, cannot be fractional.

**Reply by Ongjming** [reply]
I see, hmm I know where I went wrong now :) Once again thanks so much to both of you!

**Reply by Ongjming** [reply]
~~Sorry just came across yet another thing related to this as well DX~~
~~The vanilla Currency.asset has 7 items within it. Last time I tested with no issues, the .asset had 5 items in it.~~
~~This time round, I placed 19 items within it and the game doesn't seemed to be able to read it somehow.~~

~~So erm new question:~~
~~Is there a hardcoded limit for the amount of types of items in there before the game can no longer read it and gives the error?~~
~~(I am assuming there would be one due to what I had observed + it make sense logically that there won't be as much space to graphically represent them via icon.)~~

EDIT: I realised the error could pop up the same way with any error/mistakes made in the typing of whatever is inside the .asset file. This is a good learning point for me. Thanks!
Note to self: need to take care of the open and closing symbols.
Learning point: Everything can be used as currency and any values can be assigned (just not decimals).
Anything wrong within the file, in the error log it would display : Failed to Analyze (File location): Object reference not set to an instance of an object
